model_api/cpp/models/src/segmentation_model.cpp

```diff
@@ -172,20 +172,22 @@ void SegmentationModel::prepareInputsOutputs(std::shared_ptr<ov::Model>& model)
         throw std::logic_error("3-channel 4-dimensional model's input is expected");
     }
 
-    ov::preprocess::PrePostProcessor ppp(model);
-    ppp.input().tensor().set_element_type(ov::element::u8).set_layout({"NHWC"});
-
-    if (useAutoResize) {
-        ppp.input().tensor().set_spatial_dynamic_shape();
-
-        ppp.input()
-            .preprocess()
-            .convert_element_type(ov::element::f32)
-            .resize(ov::preprocess::ResizeAlgorithm::RESIZE_LINEAR);
+    if (!embedded_processing) {
+        model = ImageModel::embedProcessing(model,
+                                        inputNames[0],
+                                        inputLayout,
+                                        resizeMode,
+                                        interpolationMode,
+                                        ov::Shape{inputShape[ov::layout::width_idx(inputLayout)],
+                                                  inputShape[ov::layout::height_idx(inputLayout)]});
+
+        ov::preprocess::PrePostProcessor ppp = ov::preprocess::PrePostProcessor(model);
+        ppp.output().tensor().set_element_type(ov::element::f32);
+        model = ppp.build();
+        useAutoResize = true; // temporal solution for classification
```

Comment:
It is possible to ignore `useAutoResize` here. But `ImageModel` introduces that and we can't remove this because there are still old wrappers relying on that. So just to keep the state of `ImageModel` correct I set `useAutoResize` here